sporkfish/engine.py

```diff
@@ -39,8 +46,32 @@ def best_move(self, board: Board, timeout: Optional[float] = None) -> chess.Move
         :return: The selected chess move.
         :rtype: chess.Move
         """
-        opening_move = self._opening_book.query(board)
-        return opening_move or self._searcher.search(board, timeout)[1]
+        if opening_move := self._opening_book.query(board):
+            return opening_move
+
+        end_move = self._use_endgame_tablebase(board)
+        return end_move or self._searcher.search(board, timeout)[1]
+
+    def _use_endgame_tablebase(self, board: Board) -> Optional[chess.Move]:
+        """
+        Return endgame tablebase move if applicable.
+
+        :param board: The current chess board position.
+        :type board: Board
+        :return: The selected chess move.
+        :rtype: Optional[chess.Move]
+        """
+        counter = 0
+        for square in chess.SQUARES:
```

Comment:
No perf test yet. 

Next on the list for me, Tonight.

Reply:
Nice nice - look forward to seeing the results.

Reply:
On second thought I think it will be only a small hit - this function is called every time we search, with no recursion. That means it should be exactly equivalent to the number of moves we make, i.e. < 100. Given that we at maximum search 64 squares each time, this means piece_at calls are bounded (<= 64 * num_moves_made). Lets say we have num_moves_made = 100, so piece_at is called 6400 times. This is tiny, recall that in the MVV-LVA improvement we reduced the piece_at calls with order of magnitude order 10^6 (?), but only with <1.5 second improvement. Anyhow, I'll check.

Reply:
Very fair point - no performance testing was done yesterday - will leave with Kelvin!

Reply:
------------------------------------------------------------------------------------------------
         4450 function calls (3865 primitive calls) in 0.004 seconds

------------------------------------------------------------------------------------------------
         752 function calls in 0.000 seconds

------------------------------------------------------------------------------------------------
         2640 function calls (2055 primitive calls) in 0.002 seconds

3 positions, endgame query time (inclusive of checkin all squares)

Reply:
Sanity checking: the order of magnitude estimation for number of function calls here is:

```
100 (number of remaining moves from one side given a board) * 64 (number of squares on board) = 6400? 
```

Where 100 is roughly the number of moves made in a typical chess game from one side?

Reply:
Yes, exactly. Note that function calls above mean the total function calls, not just checking squares (so internal function calls due to probing will be included). Conclusion is we're good.